undofx/src/main/java/org/fxmisc/undo/impl/UndoManagerImpl.java

```diff
@@ -68,16 +70,32 @@ protected boolean computeValue() {
     private QueuePosition mark;
     private C expectedChange = null;
 
+    /**
+     * Creates an {@link UndoManager}.
+     *
+     * @param queue the queue that stores the changes to undo/redo
+     * @param invert inverts the change
+     * @param apply applies the change (undo/redo)
+     * @param merge merges two changes into one change. If the resulting change is an identity change, the two
+     *              changes will not be merged.
+     * @param isIdentity determines whether change is an identity change (e.g. {@link Function#identity()}). For
+     *                   example, {@code 0} is the identity change in
+     *                   {@code BiFunction<Integer, Integer, Integer> plus = (i, j) -> i + j} because
+     *                   {@code 4 == 4 + 0 == plus.apply(4, 0)}
+     * @param changeSource the {@link EventStream} that emits changes
+     */
     public UndoManagerImpl(
             ChangeQueue<C> queue,
             Function<? super C, ? extends C> invert,
             Consumer<C> apply,
             BiFunction<C, C, Optional<C>> merge,
+            Predicate<C> isIdentity,
             EventStream<C> changeSource) {
         this.queue = queue;
         this.invert = invert;
         this.apply = apply;
         this.merge = merge;
+        this.isNotIdentity = isIdentity.negate();
```

Comment:
Yeah, I was wondering about that. Sometimes, I think it's easier to read a `notCondition` than a `!condition` because the latter's mark is easier to miss.